tests/cli-misc.rs

```diff
@@ -394,6 +404,61 @@ fn rustup_failed_path_search() {
     });
 }
 
+
+#[test]
+fn rustup_failed_path_search_toolchain() {
+    setup(&|config| {
+        use std::env::consts::EXE_SUFFIX;
+
+        let ref rustup_path = config.exedir.join(&format!("rustup{}", EXE_SUFFIX));
+        let ref tool_path = config.exedir.join(&format!("fake_proxy{}", EXE_SUFFIX));
+        utils::hardlink_file(rustup_path, tool_path).expect("Failed to create fake proxy for test");
+
+        expect_ok(
+            config,
+            &[
+                "rustup",
+                "toolchain",
+                "link",
+                "custom-1",
+                &config.customdir.join("custom-1").to_string_lossy(),
+            ],
+        );
+
+        expect_ok(
+            config,
+            &[
+                "rustup",
+                "toolchain",
+                "link",
+                "custom-2",
+                &config.customdir.join("custom-2").to_string_lossy(),
+            ],
+        );
+
+        expect_ok(
+            config,
+            &[
+                "rustup",
+                "default",
+                "custom-2"
+            ],
+        );
+
+        let broken = &["rustup", "run", "custom-1", "fake_proxy"];
+        expect_err(
+            config,
+            broken,
+            &format!(
+                "rustup component add fake_proxy{} --toolchain custom-2",
```

Comment:
It seems that `expect_err` doesn't read second line of the `stderr`. Is this standard library bug? How to test second line? I'm not sure.

Reply:
It certainly *should* be reading all of stdout: https://github.com/rust-lang/rustup.rs/blob/416f4e98d45ed3ebba85f2347fbd383a0400d0d6/src/rustup-mock/src/clitools.rs#L358

Reply:
I believe the situation here is simply that `component_for_bin()` is returning None for your `fake_proxy` and as such no installation suggestion can be generated.  You might have to construct a fake for a proxy whose name is known to `component_for_bin()` such as `cargo-miri`

Reply:
@kinnison I will try to fix test on the following days.